std/encoding.d

```diff
@@ -796,6 +807,235 @@ template EncoderInstance(CharType : Latin1Char)
+    {
+        assert(canEncode(c));
+    }
+    body
+    {
+        return 1;
+    }
+
+    void encodeViaWrite()(dchar c)
+    {
+        if (c < 0xA0) {}
+        else if (c >= 0xFFFD) { c = '?'; }
+        else
+        {
+            ptrdiff_t n = -1;
+            foreach (i, wchar d; charMap)
+            {
+                if (c == d)
+                {
+                    n = i;
+                    break;
+                }
+            }
+            c = n == -1 ? '?' : 0xA0 + cast(dchar) n;
+        }
+
+        write(cast(Latin2Char)c);
+    }
+
+    void skipViaRead()()
+    {
+        read();
+    }
+
+    dchar decodeViaRead()()
+    {
+        Latin2Char c = read();
+        return (c >= 0xA0) ? charMap[c-0xA0] : c;
+    }
+
+    dchar safeDecodeViaRead()()
+    {
+        Latin2Char c = read();
+        return (c >= 0xA0) ? charMap[c-0xA0] : c;
+    }
+
+    dchar decodeReverseViaRead()()
+    {
+        Latin2Char c = read();
+        return (c >= 0xA0) ? charMap[c-0xA0] : c;
+    }
+
+    @property EString replacementSequence()
+    {
+        return cast(EString)("?");
+    }
+
+    mixin EncoderFunctions;
+}
+
+//=============================================================================
+//          WINDOWS-1250
+//=============================================================================
+
+/** Defines a Windows1250-encoded character. */
+enum Windows1250Char : ubyte { init }
+/**
+Defines an Windows1250-encoded string (as an array of $(D
+immutable(Windows1250Char))).
+ */
+alias Windows1250String = immutable(Windows1250Char)[]; ///
+
+template EncoderInstance(CharType : Windows1250Char)
+{
+    alias E = Windows1250Char;
+    alias EString = Windows1250String;
+
+    @property string encodingName()
+    {
+        return "windows-1250";
+    }
+
+    immutable wstring charMap =
+        "\u20ac\uFFFD\u201a\uFFFD\u201e\u2026\u2020\u2021"~
+        "\uFFFD\u2030\u0160\u2039\u015a\u0164\u017d\u0179"~
+        "\uFFFD\u2018\u2019\u201c\u201d\u2022\u2013\u2014"~
+        "\uFFFD\u2122\u0161\u203a\u015b\u0165\u017e\u017a"~
+        "\u00a0\u02c7\u02d8\u0141\u00a4\u0104\u00a6\u00a7"~
+        "\u00a8\u00a9\u015e\u00ab\u00ac\u00ad\u00ae\u017b"~
+        "\u00b0\u00b1\u02db\u0142\u00b4\u00b5\u00b6\u00b7"~
+        "\u00b8\u0105\u015f\u00bb\u013d\u02dd\u013e\u017c"~
+        "\u0154\u00c1\u00c2\u0102\u00c4\u0139\u0106\u00c7"~
+        "\u010c\u00c9\u0118\u00cb\u011a\u00cd\u00ce\u010e"~
+        "\u0110\u0143\u0147\u00d3\u00d4\u0150\u00d6\u00d7"~
+        "\u0158\u016e\u00da\u0170\u00dc\u00dd\u0162\u00df"~
+        "\u0155\u00e1\u00e2\u0103\u00e4\u013a\u0107\u00e7"~
+        "\u010d\u00e9\u0119\u00eb\u011b\u00ed\u00ee\u010f"~
+        "\u0111\u0144\u0148\u00f3\u00f4\u0151\u00f6\u00f7"~
+        "\u0159\u016f\u00fa\u0171\u00fc\u00fd\u0163\u02d9"
+    ;
+
+    bool canEncode(dchar c)
+    {
+        if (c < 0x80) return true;
+        if (c >= 0xFFFD) return false;
+        foreach(wchar d;charMap) { if (c == d) return true; }
```

Comment:
Hi, I've got finally back to this.
I am not sure if AAs are the best way to solve this? My plan was to use AA for canEncode and encode methods and current char array for decode methods.

I tried some simple testing with AAs (http://pastebin.com/U1yvfPLM)
With these results (all release with dub):
DMD (2.067):
Array:  1 sec, 764 ms, 912 μs, and 1 hnsec
AA:     294 ms, 69 μs, and 2 hnsecs

LDC2 (0.15.1):
Array:  1 hnsec
AA:     253 ms, 652 μs, and 1 hnsec

GDC (4.8.4):
Array:  0 hnsecs
AA:     238 ms, 871 μs, and 5 hnsecs

I guess that the problem is with current AA implementation so implement using AA and hope for the future AA reimplementation? Thx for advice.